canal/dump.go

```diff
@@ -99,10 +101,17 @@ func (c *Canal) AddDumpIgnoreTables(db string, tables ...string) {
 
 func (c *Canal) dump() error {
 	if c.dumper == nil {
-		return errors.New("mysqldump is not exist")
+		return errors.New("mysqldump does not exist")
 	}
 
 	h := &dumpParseHandler{c: c}
+	if c.master.GTIDSet() != nil {
```

Comment:
any useful expect checking `h.gset != nil`?

Reply:
if c.master.GTIDSet() is nil, gtid is disabled, so we should not get gtid position from master.

otherwise, it is empty at this point. The user wants to use gtid, but the current gtid position is empty. We should get it from master.